edge-modules/load-gen/Settings.cs

```diff
@@ -23,47 +25,52 @@ public class Settings
 
                 return new Settings(
                     configuration.GetValue("messageFrequency", TimeSpan.FromMilliseconds(20)),
-                    configuration.GetValue<double>("jitterFactor", 0.5),
-                    configuration.GetValue("twinUpdateFrequency", TimeSpan.FromMilliseconds(500)),
                     configuration.GetValue<ulong>("messageSizeInBytes", 1024),
-                    configuration.GetValue<TransportType>("transportType", TransportType.Amqp_Tcp_Only),
-                    configuration.GetValue<string>("outputName", "output1"));
+                    configuration.GetValue("transportType", TransportType.Amqp_Tcp_Only),
+                    configuration.GetValue("outputName", "output1"),
+                    configuration.GetValue("startDelay", TimeSpan.FromSeconds(2)));
             });
 
         Settings(
             TimeSpan messageFrequency,
-            double jitterFactor,
-            TimeSpan twinUpdateFrequency,
             ulong messageSizeInBytes,
             TransportType transportType,
-            string outputName)
+            string outputName,
+            TimeSpan startDelay)
         {
-            this.MessageFrequency = messageFrequency;
-            this.JitterFactor = jitterFactor;
-            this.TwinUpdateFrequency = twinUpdateFrequency;
-            this.MessageSizeInBytes = messageSizeInBytes;
+            this.MessageFrequency = Preconditions.CheckNotNull(messageFrequency);
+            this.MessageSizeInBytes = Preconditions.CheckRange<ulong>(messageSizeInBytes, 1);
+            this.OutputName = Preconditions.CheckNonWhiteSpace(outputName, nameof(outputName));
+            this.StartDelay = startDelay;
             this.TransportType = transportType;
-            this.OutputName = outputName;
         }
 
         public static Settings Current => DefaultSettings.Value;
 
         public TimeSpan MessageFrequency { get; }
 
-        public double JitterFactor { get; }
-
-        public TimeSpan TwinUpdateFrequency { get; }
-
         public ulong MessageSizeInBytes { get; }
 
         [JsonConverter(typeof(StringEnumConverter))]
         public TransportType TransportType { get; }
 
         public string OutputName { get; }
 
+        public TimeSpan StartDelay { get; }
+
         public override string ToString()
         {
-            return JsonConvert.SerializeObject(this, Formatting.Indented);
+            // serializing in this pattern so that secrets don't accidentally get added anywhere in the future
+            var fields = new Dictionary<string, string>()
```

Comment:
nit: I think it is not necessary to serialize as json, just build a string with the fields